controllers/new-page-pane.js

```diff
@@ -1,31 +1,192 @@
-const edit = require('../services/edit'),
-  progress = require('../services/progress');
+var edit = require('../services/edit'),
+  dom = require('@nymag/dom'),
+  keyCode = require('keycode'),
+  _ = require('lodash'),
+  progress = require('../services/progress'),
+  pane = require('../services/pane');
 
-module.exports = function () {
-  function constructor() {
+/**
+ * filter items in the list, based on the label or name
+ * @param {string} val
+ * @param {NodeList} items
+ */
+function filter(val, items) {
+  _.each(items, function (item) {
+    var label = item.textContent.toLowerCase(),
+      name = item.getAttribute('data-item-name').toLowerCase();
+
+    val = val.toLowerCase();
+
+    if (_.includes(label, val) || _.includes(name, val)) {
+      item.classList.remove('filtered');
+    } else {
+      item.classList.add('filtered');
+    }
+  });
+}
+
+/**
+ * get the available items
+ * @param {NodeList} items
+ * @returns {array}
+ */
+function getAvailable(items) {
+  return _.filter(items, (item) => !item.classList.contains('filtered'));
+}
+
+/**
+ * focus first available item in list
+ * @param {NodeList} list
+ */
+function focusFirst(list) {
+  list[0].focus();
```

Comment:
I think this'll throw an error if list[0] doesn't exist you should check for _isArray(list) maybe